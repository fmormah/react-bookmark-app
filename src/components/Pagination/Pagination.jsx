import React from "react";

const Pagination = ({ linksPerPage, totalLinks, currentPage, handlePageChange }) => {
  const pageNumbers = [];
  for (let i = 1; i <= Math.ceil(totalLinks / linksPerPage); i++) {
    pageNumbers.push(i);
  }

  return ( (totalLinks > 0 ) &&
    <nav className="pagination-navigation">
      <button
        disabled={currentPage === 1}
        data-testid="previous-button"
        onClick={() => handlePageChange(currentPage - 1)}
        className="previous-button"
      />
      {pageNumbers.map((number)=>{
         return <button className={`dot-button ${currentPage === number? 'is-active':''}`} key={number} onClick={() => handlePageChange(number)}>
          {number}
        </button>
      })}
      <button
        disabled={currentPage === pageNumbers.length}
        data-testid="next-button"
        onClick={() => handlePageChange(currentPage + 1)}
        className="next-button"
      />
    </nav>
  );
}

export default Pagination;
