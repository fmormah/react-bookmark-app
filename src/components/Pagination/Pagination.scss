.pagination-navigation {
  .previous-button,
  .next-button {
    width: 40px;
    height: 40px;
    margin: 30px 10px 0;
    border-radius: 50px;
    border: none;
    background-color: black;
    background-repeat: no-repeat;
    background-position: center;
    background-size: 25px;
    transition-duration: 0.5s;
    position: relative;
    cursor: pointer;
    left: 0;
    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='%23ffffff' width='800px' height='800px' viewBox='0 0 200 200' data-name='Layer 1' id='Layer_1'%3E%3Ctitle/%3E%3Cpath d='M100,15a85,85,0,1,0,85,85A84.93,84.93,0,0,0,100,15Zm0,150a65,65,0,1,1,65-65A64.87,64.87,0,0,1,100,165ZM97.5,57.5a9.9,9.9,0,0,0-14,14L112,100,83.5,128.5a9.9,9.9,0,0,0,14,14L126,114a19.92,19.92,0,0,0,0-28.5Z'/%3E%3C/svg%3E");
    &:disabled {
      opacity: 0.2;
    }
    &:hover {
      left: 5px;
    }
  }
  .previous-button {
    transform: scale(-1);
    &:hover {
      left: -5px;
    }
  }
  .dot-button {
    color: #fff;
    background: #000;
    border: none;
    border-radius: 100%;
    padding: 5px;
    width: 30px;
    height: 30px;
    margin: 5px;
    position: relative;
    top: -12px;
    transform: scale(0.85);
    transition-duration: 0.2s;
    border: 1px solid #000;
    cursor: pointer;
    &.is-active {
      border: 2px solid #fff;
    }
    &:hover {
      transform: scale(1.1);
    }
  }
}

.page-mobile-view {
  display: none;
  @media screen and (max-width: $mediaBreakPoint) {
    display: block;
  }
}
