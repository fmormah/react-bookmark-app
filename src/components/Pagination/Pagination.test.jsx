import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Pagination from "./Pagination";

describe("Pagination component", () => {
  const handlePageChange = jest.fn();
  const linksPerPage = 2;
  const totalLinks = 6;
  const currentPage = 1;

  it("renders correctly with totalLinks greater than 0", () => {
    const { container } = render(
      <Pagination
        linksPerPage={linksPerPage}
        totalLinks={totalLinks}
        currentPage={currentPage}
        handlePageChange={handlePageChange}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("does not render if totalLinks is 0 or less", () => {
    const { container } = render(
      <Pagination
        linksPerPage={linksPerPage}
        totalLinks={0}
        currentPage={currentPage}
        handlePageChange={handlePageChange}
      />
    );
    expect(container).toBeEmptyDOMElement();
  });

  it("calls handlePageChange on next button click", () => {
    const { getByTestId } = render(
      <Pagination
        linksPerPage={linksPerPage}
        totalLinks={totalLinks}
        currentPage={currentPage}
        handlePageChange={handlePageChange}
      />
    );
    fireEvent.click(getByTestId("next-button"));
    expect(handlePageChange).toHaveBeenCalledWith(currentPage + 1);
  });
});