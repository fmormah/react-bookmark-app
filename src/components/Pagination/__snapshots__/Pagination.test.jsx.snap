// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders correctly with totalLinks greater than 0 1`] = `
<div>
  <nav
    class="pagination-navigation"
  >
    <button
      class="previous-button"
      data-testid="previous-button"
      disabled=""
    />
    <button
      class="dot-button is-active"
    >
      1
    </button>
    <button
      class="dot-button "
    >
      2
    </button>
    <button
      class="dot-button "
    >
      3
    </button>
    <button
      class="next-button"
      data-testid="next-button"
    />
  </nav>
</div>
`;
